Feat/support embark config js (#2277)
embarklabs/embark: site/source/docs/configuration.md

}
```

> Alternatively, it's also possible to use a `embark.config.js` file instead, which exports a function that calculates the embark configuration object. This is particularly useful when the configuration needs to be built based on asynchronous operations. To make use of `embark.config.js`, simply create the file in your DApp and make sure it exports an (async) function which resolves with a config object like this:

Reviewer: ```suggestion
Alternatively, it's also possible to use an `embark.config.js` file instead, which exports a function that calculates the embark configuration object. This is particularly useful when the configuration needs to be built based on asynchronous operations. To make use of `embark.config.js`, simply create the file in your DApp and make sure it exports an (async) function which resolves with a config object like this:
```

```js
// embark.config.js

module.exports = async function () {
  const secrets = await getSecrets();
  return {
    // Embark configuration goes here
  };
};
```

> Embark will import and run the exported function inside an `async/await` context. It's also possible to return the configuration object wrapped inside a`Promise`.

Reviewer: ```suggestion
Embark will import and run the exported function inside an `async/await` context. It's also possible to return the configuration object wrapped inside a `Promise` or to just return an object. You can use this function syntax even if everything is synchronous.
```

Reviewer: Still missing the space between the `a` and `Promise`

}
```

Alternatively, it's also possible to use a `embark.config.js` file instead, which exports a function that calculates the embark configuration object. This is particularly useful when the configuration needs to be built based on asynchronous operations. To make use of `embark.config.js`, simply create the file in your DApp and make sure it exports an (async) function which resolves with a config object like this:

```js
// embark.config.js

module.exports = async function () {
  return new Promise(resolve => {
    setTimeout(() => {
      resolve({
        // configuration goes here
      })
    }, 5000);
  });
};

> ```

Reviewer: Using a timeout seems weird. Maybe we should show something simpler like: 
```
module.exports = async function () {
  const secrets = await functionToGetSecrets();
  return {
    // Embark configuration goes here
    // You can pass it your secrets and even use environment variables using `process.env`
  }
};
```

> Embark will import and run the exported functoin inside an `async/await` context. So it's also possible to simply return the configuration object without wrapping it inside a `Promise`. Notice that the `setTimeout()` in the snippet above is just for demonstrating asynchronisity.

Reviewer: ```suggestion
Embark will import and run the exported function inside an `async/await` context. So it's also possible to simply return the configuration object without wrapping it inside a `Promise`. Notice that the `setTimeout()` in the snippet above is just for demonstrating asynchronicity.
```

Reviewer: If you change the snippet like I suggested, this whole paragraph will need to be changed
